core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java

```diff
@@ -427,15 +437,54 @@ protected void finalize() throws Throwable {
 				F renamed = FileSystemUtils.renameFile((IWritableFileSystem<F>)fileSystem, source, destination, isOverwrite(), getNumberOfBackups());
 				return fileSystem.getName(renamed);
 			} else if (action.equalsIgnoreCase(ACTION_MOVE)) {
-				F file=getFile(input, pvl);
 				String destinationFolder = determineDestination(pvl);
-				F moved = FileSystemUtils.moveFile(fileSystem, file, destinationFolder, isOverwrite(), getNumberOfBackups(), isCreateFolder());
-				return fileSystem.getName(moved);
+				if(StringUtils.isNotEmpty(getWildCard()) || StringUtils.isNotEmpty(getExcludeWildCard())) { 
+					String folder = arrangeFolder(determineInputFoldername(input, pvl));
+					XmlBuilder dirXml = new XmlBuilder("MovedFilesList");
+					try(Stream<F> stream = FileSystemUtils.getFilteredList(fileSystem, folder, getWildCard(), getExcludeWildCard())) {
+						int count = 0;
+						Iterator<F> it = stream.iterator();
+						while(it.hasNext()) {
+							F file = it.next();
+							F movedFile = FileSystemUtils.moveFile(fileSystem, file, destinationFolder, isOverwrite(), getNumberOfBackups(), isCreateFolder());
+							if(movedFile != null) {
+								dirXml.addSubElement(getFileAsXmlBuilder(movedFile, "file"));
+								count++;
+							}
+						}
+						dirXml.addAttribute("count", count);
+					}
+					return dirXml.toXML();
+				} else {
+					F file=getFile(input, pvl);
+					F moved = FileSystemUtils.moveFile(fileSystem, file, destinationFolder, isOverwrite(), getNumberOfBackups(), isCreateFolder());
+					return fileSystem.getName(moved);
```

Comment:
This block is repeated several times, each time with a different operation on the file.
It would be good to turn this into a  method, where you can pass the action as a method parameter.